public/app/features/alerting/unified/components/mute-timings/useMuteTimings.tsx

```diff
@@ -156,14 +155,18 @@ export const useCreateMuteTiming = ({ alertmanager }: BaseAlertmanagerArgs) => {
 export const useGetMuteTiming = ({ alertmanager, name: nameToFind }: BaseAlertmanagerArgs & { name: string }) => {
   const useK8sApi = shouldUseK8sApi(alertmanager);
 
-  const [getGrafanaTimeInterval, k8sResponse] = useLazyReadNamespacedTimeIntervalQuery({
+  const [getGrafanaTimeInterval, k8sResponse] = useLazyListNamespacedTimeIntervalQuery({
     selectFromResult: ({ data, ...rest }) => {
       if (!data) {
         return { data, ...rest };
       }
 
+      if (data.items.length === 0) {
+        return { ...rest, data: undefined, isError: true };
```

Comment:
I'm not sure if I understand the intention here; why do we assume an error occurred if there are no time intervals?

Reply:
This is here as we're using the list endpoint as a substitute to find a specific interval - and if we've filtered down the list and there are none returned, then we're looking for something that doesn't exist. Without this, the detail view for a time interval wouldn't correctly go into the error state for not being able to find the interval. You can test this by misspelling the name of an interval and checking it shows the error